py3status/modules/imap.py

```diff
@@ -98,28 +126,52 @@ def _connection_starttls(self):
         connection.starttls(create_default_context())
         return connection
 
+    def _disconnect(self):
+        try:
+            if self.connection is not None:
+                self.connection.close()
+                self.connection.logout()
+        finally:
+            self.connection = None
+
     def _get_mail_count(self):
         try:
-            mail_count = 0
-            directories = self.mailbox.split(',')
-
-            if self.security == "ssl":
-                connection = self._connection_ssl()
-            elif self.security == "starttls":
-                connection = self._connection_starttls()
-
-            connection.login(self.user, self.password)
-
-            for directory in directories:
-                connection.select(directory)
-                unseen_response = connection.search(None, self.criterion)
-                mails = unseen_response[1][0].split()
-                mail_count += len(mails)
-
-            connection.close()
-            return mail_count
-        except:
-            return None
+            while True:
+                if self.connection is None:
+                    if self.security == "ssl":
+                        self.connection = self._connection_ssl()
+                    elif self.security == "starttls":
+                        self.connection = self._connection_starttls()
+
+                    self.connection.login(self.user, self.password)
+
+                tmp_mail_count = 0
+                directories = self.mailbox.split(',')
+
+                for directory in directories:
+                    self.connection.select(directory)
+                    unseen_response = self.connection.search(None, self.criterion)
+                    mails = unseen_response[1][0].split()
+                    tmp_mail_count += len(mails)
+
+                self.mail_count = tmp_mail_count
+
+                if self.use_idle:
+                    self.connection.idle()
+                else:
+                    return
+        except socket_error as e:
+            self.mail_error = "Socket error - " + e
+            self.connection = None
```

Comment:
```
1d [chris:~/src/py3status/py3status/modules] girst-master+ ± python3 imap.py
must be str, not gaierror (TypeError) module.py line 729.
Traceback
TypeError: must be str, not gaierror
  File "/home/chris/src/py3status/py3status/module.py", line 729, in run
    response = method()
  File "imap.py", line 99, in check_mail
    self._get_mail_count()
  File "imap.py", line 162, in _get_mail_count
    self.mail_error = "Socket error - " + e
```

Reply:
yeah...I spoke too soon :| expect the fix in an hour or two. I'll also make it not error out, but continue with a warning.